provider/core/src/request.rs

```diff
@@ -371,6 +371,24 @@ impl DataLocale {
         self == <&DataLocale>::default()
     }
 
+    /// Returns an ordering suitable for use in [`BTreeSet`].
+    ///
+    /// The ordering may or may not be equivalent to string ordering, and it
+    /// may or may not be stable across ICU4X releases.
+    ///
+    /// [`BTreeSet`]: alloc::collections::BTreeSet
+    pub fn total_cmp(&self, other: &Self) -> Ordering {
```

Comment:
remind me why `DataLocale` is not `Ord`?

Reply:
Same reason `Locale` is not `Ord`, because there are multiple possible orderings and we couldn't agree on which ordering to use by default.